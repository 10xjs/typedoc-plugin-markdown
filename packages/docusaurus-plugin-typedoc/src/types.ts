export interface PluginOptions {
  inputFiles: string[];
  docsRoot: string;
  out: string;
  sidebar: SidebarOptions | null;
  plugin: string[];
  disableOutputCheck?: boolean;
}

export interface SidebarOptions {
  fullNames: boolean;
  sidebarFile: string;
  globalsLabel: string;
  readmeLabel: string;
}

export interface Sidebar {
  [sidebarId: string]: SidebarItem[];
}

export interface SidebarCategory {
  type: string;
  label: string;
  items: SidebarItem[];
}

export type SidebarItem = SidebarCategory | string;

export interface LoadedContent {
  navigation: any;
}
