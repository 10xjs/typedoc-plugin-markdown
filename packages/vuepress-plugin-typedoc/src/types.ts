import { Application, ProjectReflection } from 'typedoc';

export interface PluginOptions {
  id?: string;
  logger?: string;
  inputFiles?: string[];
  out?: string;
  hideBreadcrumbs: boolean;
  hideProjectName: boolean;
  sidebar?: SidebarOptions | null;
}

export interface SidebarOptions {
  fullNames: boolean;
  parentCategory: string;
}

export interface LoadedContent {
  app: Application;
  project: ProjectReflection;
}
