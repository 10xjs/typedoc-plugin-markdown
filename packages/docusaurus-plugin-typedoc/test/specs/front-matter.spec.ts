import cuid from 'cuid';
import { PageEvent } from 'typedoc/dist/lib/output/events';

import { TestApp } from '../../../typedoc-plugin-markdown/test/test-app';
import { DocsaurusFrontMatterComponent } from '../../dist/components/front-matter.component';
import { SidebarOptions } from '../../dist/types';

function generate(testApp: TestApp, sidebar: SidebarOptions) {
  testApp.bootstrap();
  const componentNamename = cuid();
  testApp.renderer.addComponent(
    componentNamename,
    new DocsaurusFrontMatterComponent(testApp.renderer, 'api', sidebar),
  );
  const frontMatterComponent = testApp.renderer.getComponent(
    componentNamename,
  ) as DocsaurusFrontMatterComponent;
  return frontMatterComponent;
}

function getPage(testApp: TestApp) {
  const reflection = testApp.findReflection('FrontMatterClass');
  const page = {
    project: testApp.project,
    model: reflection,
    url: reflection.url,
    navigation: testApp.renderer.theme.getNavigation(testApp.project),
    contents: 'CONTENTS',
  } as PageEvent;
  return page;
}

describe(`FrontMatter:`, () => {
  let testApp: TestApp;
  beforeAll(() => {
    testApp = new TestApp(['frontmatter.ts']);
  });

  test(`should return front with short names`, () => {
    const frontMatterComponent = generate(testApp, {
      fullNames: false,
      sidebarFile: 'typedoc-sidebar.js',
      globalsLabel: 'Globals',
      readmeLabel: 'README',
    });
    const page = getPage(testApp);
    frontMatterComponent.onPageEnd(page);
    expect(page.contents).toMatchSnapshot();
  });

  test(`should return front matter with full names`, () => {
    const frontMatterComponent = generate(testApp, {
      fullNames: true,
      sidebarFile: 'typedoc-sidebar.js',
      globalsLabel: 'Globals',
      readmeLabel: 'README',
    });
    const page = getPage(testApp);
    frontMatterComponent.onPageEnd(page);
    expect(page.contents).toMatchSnapshot();
  });

  test(`should return front matter without sidebar`, () => {
    const frontMatterComponent = generate(testApp, null);
    const page = getPage(testApp);
    frontMatterComponent.onPageEnd(page);
    expect(page.contents).toMatchSnapshot();
  });

  test(`should set slug for index page`, () => {
    const frontMatterComponent = generate(testApp, null);
    const page = {
      project: testApp.project,
      model: testApp.project,
      url: 'index.md',
      navigation: testApp.renderer.theme.getNavigation(testApp.project),
      contents: 'CONTENTS',
    } as PageEvent;
    frontMatterComponent.onPageEnd(page);
    expect(page.contents).toMatchSnapshot();
  });
});
