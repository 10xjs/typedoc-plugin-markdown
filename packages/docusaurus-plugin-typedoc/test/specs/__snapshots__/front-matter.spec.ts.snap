// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FrontMatter: should return front matter with full names 1`] = `
"---
id: \\"_frontmatter_.frontmatterclass\\"
title: \\"Class: FrontMatterClass\\"
sidebar_label: \\"frontmatter.FrontMatterClass\\"
hide_title: true
hide_table_of_contents: true
---

CONTENTS"
`;

exports[`FrontMatter: should return front matter without sidebar 1`] = `
"---
id: \\"_frontmatter_.frontmatterclass\\"
title: \\"Class: FrontMatterClass\\"
hide_title: true
hide_table_of_contents: true
---

CONTENTS"
`;

exports[`FrontMatter: should return front with short names 1`] = `
"---
id: \\"_frontmatter_.frontmatterclass\\"
title: \\"Class: FrontMatterClass\\"
sidebar_label: \\"FrontMatterClass\\"
hide_title: true
hide_table_of_contents: true
---

CONTENTS"
`;
